pandas/core/algorithms.py

```diff
@@ -104,6 +104,12 @@ def _ensure_data(values, dtype=None):
             dtype = values.dtype
         else:
             # Datetime
+            if values.ndim > 1:
+                # Avoid calling the DatetimeIndex constructor as it is 1D only
+                asi8 = values.view('i8')
+                dtype = values.dtype
+                return asi8, dtype, 'int64'
+
             from pandas import DatetimeIndex
```

Comment:
can we *just* use the new added code you have (w/o) the ndim check and remove lines 113-115?

Reply:
Just tried it, this fails in test cases with e.g. datetime64[D] data